src/oracle_implementations/spot_price/Chainlink/LUSD3CRV/LUSD3CRVValueProvider.t.sol

```diff
@@ -0,0 +1,197 @@
+            MockProvider.ReturnData({success: true, data: abi.encode(18)}),
+            false
+        );
+
+        curveLUSD3PoolMock = new MockProvider();
+        curveLUSD3PoolMock.givenQueryReturnResponse(
+            abi.encodeWithSelector(ICurvePool.get_virtual_price.selector),
+            MockProvider.ReturnData({
+                success: true,
+                data: abi.encode(_lusd3PoolVirtualPrice)
+            }),
+            false
+        );
+
+        curveLUSD3PoolMock.givenQueryReturnResponse(
+            abi.encodeWithSelector(ICurvePool.decimals.selector),
+            MockProvider.ReturnData({success: true, data: abi.encode(18)}),
+            false
+        );
+
+        mockChainlinkLUSD = new MockProvider();
+        initChainlinkMockProvider(mockChainlinkLUSD, _lusdPrice, 8);
+
+        mockChainlinkUSDC = new MockProvider();
+        initChainlinkMockProvider(mockChainlinkUSDC, _usdcPrice, 8);
+
+        mockChainlinkDAI = new MockProvider();
+        initChainlinkMockProvider(mockChainlinkDAI, _daiPrice, 8);
+
+        mockChainlinkUSDT = new MockProvider();
+        initChainlinkMockProvider(mockChainlinkUSDT, _usdtPrice, 8);
+
+        chainlinkVP = new LUSD3CRVValueProvider(
+            // Oracle arguments
+            // Time update window
+            _timeUpdateWindow,
+            // Chainlink arguments
+            address(curve3PoolMock),
+            address(curveLUSD3PoolMock),
+            address(mockChainlinkLUSD),
+            address(mockChainlinkUSDC),
+            address(mockChainlinkDAI),
+            address(mockChainlinkUSDT)
+        );
+    }
+
+    function test_deploy() public {
+        assertTrue(address(chainlinkVP) != address(0));
+    }
+
+    function test_getValue() public {
+        // Expected value is the value sent by the mock provider in 10**18 precision
+        // The expect value was computed with this formula:
+        // solhint-disable-next-line
+        // https://www.wolframalpha.com/input?i2d=true&i=%5C%2840%29Divide%5B1012508837937838125%2CPower%5B10%2C18%5D%5D%5C%2841%29+*+minimum%5C%2840%29Divide%5B100102662%2CPower%5B10%2C8%5D%5D%5C%2844%29%5C%2840%29+Divide%5B1020628557740573240%2CPower%5B10%2C18%5D%5D+*+Divide%5Bminimum%5C%2840%29100030972%5C%2844%29100100000%5C%2844%29100000298%5C%2841%29%2CPower%5B10%2C8%5D%5D%5C%2841%29%5C%2841%29
+        int256 expectedValue = 1013548299761041868;
+        // Computed value based on the parameters that are sent via the mock provider
+        int256 value = chainlinkVP.getValue();
+
+        assertTrue(value == expectedValue);
+    }
+
+    function test_description() public {
+        string memory expectedDescription = "LUSD3CRV";
+        string memory desc = chainlinkVP.description();
+        assertTrue(
+            keccak256(abi.encodePacked(desc)) ==
+                keccak256(abi.encodePacked(expectedDescription))
+        );
+    }
+
+    function testFail_deploy_shouldFailWithInvalidLUSDPoolDecimals() public {
```

Comment:
@amarcu This is an opportunity to use the new cheatcodes Foundry has to offer.

Add the additional condition of checking the fail message by using [`expectRevert()`](https://book.getfoundry.sh/reference/cheatcodes.html#expectrevert).